### PR TITLE
Selectively support configuring and creating a kubeclient.

### LINKED ISSUE #743: Enable some commands to opt out of supporting --kubeconfig and --master flags
For instance, `riff image relocate` does not need these flags and yet it can't avoid inheriting them.

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes #743

### CODE REVIEW

File: docs/riff_system_install.md [hunk 1]
```diff
@@ -35,13 +35,6 @@ riff system install [flags]
       --node-port         whether to use NodePort instead of LoadBalancer for ingress gateways
 ```
 
-### Options inherited from parent commands
```
Comment (scothis) [inline docs/riff_system_install.md]: should still support kubeconfig

File: docs/riff_system_uninstall.md [hunk 2]
```diff
@@ -26,13 +26,6 @@ riff system uninstall [flags]
       --istio   include Istio and the istio-system namespace in the removal
 ```
 
-### Options inherited from parent commands
```
Comment (scothis) [inline docs/riff_system_uninstall.md]: should still support kubeconfig

File: docs/riff_system_install.md (hunk 1 above)
Comment (ericbottard) [reply docs/riff_system_install.md]: It actually doesn't (currently)

Comment (scothis) [reply docs/riff_system_install.md]: that's a bug, we should pass these flags to kubectl when we shell out to it.

**Comment by ericbottard** [comment]
Let's address those when we do https://github.com/projectriff/riff/issues/803

**Comment by glyn** [comment]
@ericbottard Then please could you change the FIXMEs that refer to bug 743 to refer to bug 803 *and* update bug 803 to mention this. All in all, wouldn't it be simpler to just delete the code tagged by the FIXMEs and leave bug 803 to add the new support for passing the relevant parms to kubectl?

**Comment by ericbottard** [comment]
Did the latter and force pushed a rebased version

**Comment by ericbottard** [comment]
This is ready for merge review, as far as I'm concerned. PTAL

**Comment by glyn** [comment]
Merged as c995518217ff1e470d7e62b3d3c3f3542ca5f3a3

**Pull request closed without merging**